### PR TITLE
154 pdf thumbnail preview

### PR DESCRIPTION
**Pull request by AnishDabhane**

### CODE REVIEW

File: frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx [hunk 1]
```diff
@@ -12,6 +12,11 @@ import { memo } from "react"
 import { toast } from "sonner"
 import { useIsDesktop } from "@/hooks/useMediaQuery"
 
+import path from 'path';
+import { fromPath } from 'pdf2pic';
+import fs from 'fs';
+
+import test from '../../../../../../../../../../../../Downloads/test.png'
```
Comment (janhvipatil) [inline frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx:19]: Remove the hardcoded test image import and replace it with a more appropriate placeholder or loading state.

File: package.json [hunk 2]
```diff
@@ -13,5 +13,9 @@
   },
   "keywords": [],
   "author": "",
-  "license": "AGPL-3.0-only"
-}
\ No newline at end of file
+  "license": "AGPL-3.0-only",
+  "dependencies": {
```
Comment (janhvipatil) [inline package.json:17]: move the dependencies to the package.json file inside the "frontend" folder

File: frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx [hunk 3]
```diff
@@ -54,40 +59,46 @@ export const FileMessageBlock = memo(({ message, user, ...props }: FileMessageBl
...
+                height="220px"
+                maxWidth={isDesktop ? '320px' : '100%'}
                 className="border-2 bg-gray-2 dark:bg-gray-4 rounded-md border-gray-4  dark:border-gray-6 shadow-sm">
-                <Flex align='center' gap='2'>
-                    <FileExtensionIcon ext={fileExtension} />
-                    <Text as='span' size='2' className="text-ellipsis overflow-hidden line-clamp-1">{fileName}</Text>
-                </Flex>
+            
+                            
+                <img src={test} width="100%" height="80%"/>
```
Comment (janhvipatil) [inline frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx:72]: Remove the hardcoded test image and replace it with a more appropriate placeholder or loading state

File: frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx [hunk 4]
```diff
@@ -104,6 +115,14 @@ const PDFPreviewButton = ({ message, user }: {
 
     const fileName = getFileName(message.file)
 
+    const image = convertPdfToImage(message, user)
```
Comment (janhvipatil) [inline frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx:118]: The image variable is being assigned the result of a Promise chain, not the actual converted image. The convertPdfToImage function's result is not being properly captured or returned. The then block doesn't return anything, so image will be a Promise that resolves to undefined. Also, "image" is not even being used anywhere, so I'm not sure about this implementation. Did it work for you?

File: package.json (hunk 2 above)
Comment (AnishDabhane) [reply package.json:17]: ok

File: frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx (hunk 4 above)
Comment (AnishDabhane) [reply frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx:118]: I was working on implementation, but due to time constraints, I had to create a PR . Nevertheless I'll finish the implementation over this weekend.

Comment (janhvipatil) [reply frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx:118]: okay, you can move the PR in draft mode until you are done and then re request a review.

**Pull request closed without merging**